sdl_android/src/androidTest/java/com/smartdevicelink/protocol/WiProProtocolTests.java

```diff
@@ -342,6 +344,35 @@ public void testOverallocatingAccumulator(){
 		assertNull(oom_error);
 		
 	}
+
+	/**
+	 * Test data being null in SendMessage() method
+	 *
+	 * @see WiProProtocol#SendMessage(ProtocolMessage)
+	 */
+	public void testNullDataInSendMessage() {
+		Object protocolMsg = null;
+		Field dataField = null;
+		Field msgLockField = null;
+		try {
+			protocolMsg = Class.forName("com.smartdevicelink.protocol.ProtocolMessage").newInstance();
```

Comment:
Why is reflection necessary here? ProtocolMessage is a public class with a public no argument constructor.

Reply:
I recall that for this test, without using reflection, setting the `_data` field to null for ProtocolMessage will not work because the flow would be:
```java
ProtocolMessage ptcMsg = new ProtocolMessage();
ptcMsg.setData(null); // or any public setData method
```
The follow through initialization in ProtocolMessage class will always cause a npe.
Reflection allows setting the `_data` field will null without going through the initialization.

Come to think of it now, this `_data` field is next to impossible to be null and I was wondering if we should leave this test out because it only tests if the method can handle (ignore) the case where `_data` is null.

Reply:
My question is strictly related to the constructor call. Setting the `_data` field to null can only happen through reflection for the reasons you stated. I believe the purpose of the PR is to test for a case in which `_data` is null, so if that's the case I think the test makes sense.